### ISSUE TITLE
Multiple Terms request

### ISSUE THREAD

**Issue by prvst** [HUPO-PSI/psi-ms-CV#21]
Hi;

We need terms for the following tools and their scores:

**MSFragger**

**PeptideProphet** 
- peptideprophet probability

**ProteinProphet**
- proteinprophet score
- protein group probability
- protein probability
- peptide probability
- peptide weight
- peptide group weight

**InterProphet**
- interprophet probability

**PTMProphet**
 - ptmprophet site localization predictions.

We also need a term to describe if a certain peptide was classified as razor. I noticed that some terms incorporate that into their description but they are for MaxQuant and Mascot, we need a generic term for 
 different pipelines.

Thanks

**Comment by edeutsch** [comment]
Hi Felipe, thanks for the request. For MSFragger, it looks like we indeed do need a new term. As for the other tools, they are all already in the CV under this group:
https://www.ebi.ac.uk/ols/ontologies/ms/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FMS_1002286
(Click on the + to expand the group).
Note that although the executable is named InterProphet, the tool it officially called iProphet.

As far as the probability, I have tried to resist using tool-specific probabilities and advocated generic terms. In our TPP export, we use the generic terms that you’ll find under here:
https://www.ebi.ac.uk/ols/ontologies/ms/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FMS_1002693
(again, click on + to expand)

I advocate using such terms as:
PSM-level probability ^ http://purl.obolibrary.org/obo/MS_1002357
for PeptideProphet, and:
peptide sequence-level probability ^ http://purl.obolibrary.org/obo/MS_1002362
for iProphet, etc.

i.e., I am advocating that any tool that computes a PSM-level probability (like PeptideProphet) use MS:1002357, rather than having a different term for each tool. Not everyone agrees.

So, in summary, I think most of the terms you need (except for MSFragger) are there already. We use them in our tpp2mzid exporter that can convert pepXML and protXML to mzIdentML.

**Comment by prvst** [comment]
Hi Eric;

I totally agree with having a generic term for the probabilities, that will work for us.

Thanks.

**Comment by prvst** [comment]
Did you thought about having a generic term for peptides assigned as razor ?

**Comment by germa** [comment]
Hi Felipe,

are 'peptide weight' and 'peptide group weight'
ProteinProphet input (in the sense of a threshold) or output parameters?

Cheers,
Gerhard


Am 02.04.2019 um 21:38 schrieb Felipe Leprevost:
> # peptide weight
> # peptide group weight

**Comment by prvst** [comment]
I would say its more like an output parameter, but we use it for our FDR filtering, see below an example from a ProteinProphet output. The value ranges from 0 to 1 and its present on all \<peptide\> tags

```
<protein_group group_number="1" probability="1.0000">
      <protein protein_name="NP_000009.1" n_indistinguishable_proteins="4" probability="1.0000" percent_coverage="6.9" unique_stripped_peptides="EGMAALQSDPWQQELYR+MLCDTWCIEAAAR+TSAVPSPCGK" group_sibling_id="a" total_number_peptides="15" total_number_distinct_peptides="15" pct_spectrum_ids="0.051" confidence="1.000">
         <parameter name="prot_length" value="655"/>
         <annotation protein_description="very long-chain specific acyl-CoA dehydrogenase, mitochondrial isoform 1 precursor GN=ACADVL [Homo sapiens]"/>
         <indistinguishable_protein protein_name="NP_001029031.1">
            <annotation protein_description="very long-chain specific acyl-CoA dehydrogenase, mitochondrial isoform 2 precursor GN=ACADVL [Homo sapiens]"/>
         </indistinguishable_protein>
         <indistinguishable_protein protein_name="NP_001257376.1">
            <annotation protein_description="very long-chain specific acyl-CoA dehydrogenase, mitochondrial isoform 3 GN=ACADVL [Homo sapiens]"/>
         </indistinguishable_protein>
         <indistinguishable_protein protein_name="NP_001257377.1">
            <annotation protein_description="very long-chain specific acyl-CoA dehydrogenase, mitochondrial isoform 4 GN=ACADVL [Homo sapiens]"/>
         </indistinguishable_protein>
         <peptide peptide_sequence="EGMAALQSDPWQQELYR" charge="2" initial_probability="0.9990" nsp_adjusted_probability="0.9995" fpkm_adjusted_probability="0.9995" weight="1.00" group_weight="1.00" is_nondegenerate_evidence="Y" n_enzymatic_termini="2" n_sibling_peptides="3.00" n_sibling_peptides_bin="7" n_instances="1" exp_tot_instances="1.00" is_contributing_evidence="Y" calc_neutral_pep_mass="2036.9261">
<modification_info modified_peptide="EGM[147]AALQSDPWQQELYR">
<mod_aminoacid_mass position="3" mass="147.035400"/>
</modification_info>
         </peptide>
         <peptide peptide_sequence="MLCDTWCIEAAAR" charge="2" initial_probability="0.9990" nsp_adjusted_probability="0.9995" fpkm_adjusted_probability="0.9995" weight="1.00" group_weight="1.00" is_nondegenerate_evidence="Y" n_enzymatic_termini="2" n_sibling_peptides="3.00" n_sibling_peptides_bin="7" n_instances="10" exp_tot_instances="9.99" is_contributing_evidence="Y" calc_neutral_pep_mass="1824.8523">
<modification_info mod_nterm_mass="230.1708" modified_peptide="n[230]MLCDTWCIEAAAR">
<mod_aminoacid_mass position="3" mass="160.030700"/>
<mod_aminoacid_mass position="7" mass="160.030700"/>
</modification_info>
         </peptide>
         <peptide peptide_sequence="TSAVPSPCGK" charge="2" initial_probability="0.9990" nsp_adjusted_probability="0.9995" fpkm_adjusted_probability="0.9995" weight="1.00" group_weight="1.00" is_nondegenerate_evidence="Y" n_enzymatic_termini="2" n_sibling_peptides="3.00" n_sibling_peptides_bin="7" n_instances="3" exp_tot_instances="2.98" is_contributing_evidence="Y" calc_neutral_pep_mass="1460.8062">
<modification_info mod_nterm_mass="230.1708" modified_peptide="n[230]TSAVPSPCGK">
<mod_aminoacid_mass position="8" mass="160.030700"/>
<mod_aminoacid_mass position="10" mass="357.257900"/>
</modification_info>
         </peptide>
         <peptide peptide_sequence="TSAVPSPCGK" charge="3" initial_probability="0.9990" nsp_adjusted_probability="0.9995" fpkm_adjusted_probability="0.9995" weight="1.00" group_weight="1.00" is_nondegenerate_evidence="Y" n_enzymatic_termini="2" n_sibling_peptides="3.00" n_sibling_peptides_bin="7" n_instances="1" exp_tot_instances="1.00" is_contributing_evidence="Y" calc_neutral_pep_mass="1689.9692">
<modification_info mod_nterm_mass="230.1708" modified_peptide="n[230]TS[316]AVPSPCGK">
<mod_aminoacid_mass position="2" mass="316.194900"/>
<mod_aminoacid_mass position="8" mass="160.030700"/>
<mod_aminoacid_mass position="10" mass="357.257900"/>
</modification_info>
         </peptide>
      </protein>
</protein_group>
```

**Comment by germa** [comment]
Dear proteomics/metabolomics community,

following are the new/changed terms for the release candidate 4.1.25_rc1 
of the
psi-ms.obo file.


New CV terms in version 4.1.25_rc1 of psi-ms.obo:
=================================================
[Term]
id: MS:1003011
name: pFind
def: "Sequence-tag-based search engine pFind." [PMID:30295672]
is_a: MS:1001456 ! analysis software

[Term]
id: MS:1003012
name: KSDP score
def: "Kernel mass spectral dot product scoring function." [PMID:15044235]
is_a: MS:1001143 ! PSM-level search engine specific statistic
relationship: has_order MS:1002108 ! higher score better

[Term]
id: MS:1003013
name: i3tms
def: "i3-tms search engine and data-analysis software." [PSI:PI]
is_a: MS:1001456 ! analysis software

[Term]
id: MS:1003014
name: MSFragger
def: "A database search-based peptide identification tool." [PMID:28394336]
is_a: MS:1002286 ! Trans-Proteomic Pipeline software

[Term]
id: MS:1003015
name: razor peptide
def: "A peptide which is shared between protein groups and assigned to 
the protein group with the largest number of identified peptides." [PSI:PI]
is_a: MS:1000860 ! peptide

[Term]
id: MS:1003016
name: PTMProphet:PTM Score
def: "The PTM score from PTMProphet software." [PSI:MS]
xref: value-type:xsd\:string "The allowed value-type for this CV term."
is_a: MS:1001968 ! PTM localization PSM-level statistic
relationship: has_regexp MS:1002505 ! regular expression for 
modification localization scoring

[Term]
id: MS:1003017
name: ProteinProphet:peptide weight
def: "Peptide weight used for FDR filtering." [PSI:PI]
xref: value-type:xsd\:float "The allowed value-type for this CV term."
is_a: MS:1002490 ! peptide-level scoring
is_a: MS:1001153 ! search engine specific score
relationship: has_order MS:1002108 ! higher score better
relationship: has_domain MS:1002305 ! value between 0 and 1 inclusive

[Term]
id: MS:1003018
name: ProteinProphet:peptide group weight
def: "Peptide group weight used for FDR filtering." [PSI:PI]
xref: value-type:xsd\:float "The allowed value-type for this CV term."
is_a: MS:1002490 ! peptide-level scoring
is_a: MS:1001153 ! search engine specific score
relationship: has_order MS:1002108 ! higher score better
relationship: has_domain MS:1002305 ! value between 0 and 1 inclusive


Changed CV terms in version 4.1.25_rc1 of psi-ms.obo:
=====================================================
************ added synonym 'InterProphet'
[Term]
id: MS:1002288
name: iProphet
def: "A program in the TPP that calculates distinct peptide 
probabilities based on several lines of corroborating evidence including 
search results from multiple search engines via the pepXML format." 
[PMID:21876204]
synonym: "InterProphet" EXACT []
is_a: MS:1002286 ! Trans-Proteomic Pipeline software

Best Regards,
Gerhard

-- 

----------------------------------------------------------------------

Dipl. Inform. med., Dipl. Wirtsch. Inf. GERHARD MAYER

Scientific assistant

Medizinisches Proteom-Center

DEPARTMENT Medical Bioinformatics

Building ZKF E.055 | Universitätsstraße 150 | D-44801 Bochum

Fon +49 (0)234 32-29272 | Fax +49 (0)234 32-14554

E-mail gerhard.mayer@rub.de

www.medizinisches-proteom-center.de

**Comment by prvst** [comment]
Thanks Gerhard, That looks great.

I have another question, Is there any policy preventing the creation of terms for tools not yet published ? We have been working on a pipeline that is already public and we are going to start working with mzID files now, it would be great if we could have the tool as a Term as well.

Name: Philosopher
Purpose: general proteomics processing toolkit.
Link: https://philosopher.nesvilab.org/

**Comment by prvst** [comment]
I just noticed something, on the MSFragger term, you defined the tool as been part of TPP, that is not correct. MSFragger is from Alexey Nesvizhskii's lab.

**Comment by germa** [comment]
Hi Gerhard, Felipe, thanks for this proposal, I have a few suggested
amendments, below in red (and *** in case color doesn't come through):



-----Original Message-----
From: mayerg97 via Psidev-ms-vocab <psidev-ms-vocab@lists.sourceforge.net>

Dear proteomics/metabolomics community,



following are the new/changed terms for the release candidate 4.1.25_rc1 of
the psi-ms.obo file.





New CV terms in version 4.1.25_rc1 of psi-ms.obo:

=================================================

[Term]

id: MS:1003011

name: pFind

def: "Sequence-tag-based search engine pFind." [PMID:30295672]

is_a: MS:1001456 ! analysis software



[Term]

id: MS:1003012

name: KSDP score

def: "Kernel mass spectral dot product scoring function." [PMID:15044235]

is_a: MS:1001143 ! PSM-level search engine specific statistic

relationship: has_order MS:1002108 ! higher score better



[Term]

id: MS:1003013

name: i3tms

def: "i3-tms search engine and data-analysis software." [PSI:PI]

is_a: MS:1001456 ! analysis software



[Term]

id: MS:1003014

name: MSFragger

def: "A database search-based peptide identification tool." [PMID:28394336]

is_a: MS:1002286 ! Trans-Proteomic Pipeline software



[Term]

id: MS:1003015

name: razor peptide

def: "A peptide which is shared between protein groups and assigned to

the protein group with the largest number of identified peptides." [PSI:PI]

is_a: MS:1000860 ! peptide



*** Begin

The following term should not be added as is. PTMProphet doesn’t have a
“PTM Score”.

x [Term]

x id: MS:1003016

x name: PTMProphet:PTM Score

x def: "The PTM score from PTMProphet software." [PSI:MS]

x xref: value-type:xsd\:string "The allowed value-type for this CV term."

X is_a: MS:1001968 ! PTM localization PSM-level statistic

X relationship: has_regexp MS:1002505 ! regular expression for modification
localization scoring

*** PTM Prophet has quite a substantial number of metrics.

At the PSM level, we have:

normalized information content

number of localized modifications

At the site level, we have:

               modification localization probability

               O-score

M-score

So I think we may actually want 5 terms, not a single term (and not called
PTM Score).

@Felipe, are you eager to write terms, or were you just thinking that we
will need something eventually?

The paper is currently under revision. I suggest we exclude 3016 from this
RC, and circle back in a few weeks to provide a more complete set of terms
and definitions. Does that sound okay?

*** End





[Term]

id: MS:1003017

name: ProteinProphet:peptide weight

def: "Peptide weight used for FDR filtering." [PSI:PI]

***def: "Fraction of peptide evidence attributable to a protein or a set of
indistinguishable proteins." [PSI:PI]

xref: value-type:xsd\:float "The allowed value-type for this CV term."

is_a: MS:1002490 ! peptide-level scoring

is_a: MS:1001153 ! search engine specific score

relationship: has_order MS:1002108 ! higher score better

relationship: has_domain MS:1002305 ! value between 0 and 1 inclusive

*** NOTE: Although some may use this for filtering, any score could
potentially be used for filtering, and I suggest that a potential use
should not be part of the definition, or else 100 other terms would also be
characterized as “for FDR filtering” (but I discourage that)



[Term]

id: MS:1003018

name: ProteinProphet:peptide group weight

def: "Peptide group weight used for FDR filtering." [PSI:PI]

***def: "Fraction of peptide evidence attributable to a group of proteins."
[PSI:PI]

xref: value-type:xsd\:float "The allowed value-type for this CV term."

is_a: MS:1002490 ! peptide-level scoring

is_a: MS:1001153 ! search engine specific score

relationship: has_order MS:1002108 ! higher score better

relationship: has_domain MS:1002305 ! value between 0 and 1 inclusive





Changed CV terms in version 4.1.25_rc1 of psi-ms.obo:

=====================================================

************ added synonym 'InterProphet'

[Term]

id: MS:1002288

name: iProphet

def: "A program in the TPP that calculates distinct peptide

probabilities based on several lines of corroborating evidence including

search results from multiple search engines via the pepXML format."

[PMID:21876204]

synonym: "InterProphet" EXACT []

is_a: MS:1002286 ! Trans-Proteomic Pipeline software



Best Regards,

Gerhard



-- 



----------------------------------------------------------------------



Dipl. Inform. med., Dipl. Wirtsch. Inf. GERHARD MAYER



Scientific assistant



Medizinisches Proteom-Center



DEPARTMENT Medical Bioinformatics



Building ZKF E.055 | Universitätsstraße 150 | D-44801 Bochum



Fon +49 (0)234 32-29272 | Fax +49 (0)234 32-14554



E-mail gerhard.mayer@rub.de



www.medizinisches-proteom-center.de







_______________________________________________

Psidev-ms-vocab mailing list

Psidev-ms-vocab@lists.sourceforge.net

https://lists.sourceforge.net/lists/listinfo/psidev-ms-vocab

**Comment by prvst** [comment]
@germa 
We are currently implementing the mzID format on our pipeline, that's why we need to have tose terms, but I guess we can wait a little until you have a new version.

Thanks

**Comment by germa** [comment]
Included in psi-ms.obo since version 4.1.25
